src/main/io/serial_1wire_vcp.c

```diff
@@ -0,0 +1,237 @@
+#define ESC_OUTPUT  gpioInit(escHardware[selected_esc].gpio, &escHardware[selected_esc].gpio_config_OUTPUT)
+
+void usb1WireInitializeVcp(void)
+{
+    pwmDisableMotors();
+    selected_esc = 0;
+    memset(&escHardware, 0, sizeof(escHardware));
+    pwmIOConfiguration_t *pwmIOConfiguration = pwmGetOutputConfiguration();
+    for (volatile uint8_t i = 0; i < pwmIOConfiguration->ioCount; i++) {
+      if ((pwmIOConfiguration->ioConfigurations[i].flags & PWM_PF_MOTOR) == PWM_PF_MOTOR) {
+          if(motor[pwmIOConfiguration->ioConfigurations[i].index] > 0) {
+              escHardware[selected_esc].gpio = pwmIOConfiguration->ioConfigurations[i].timerHardware->gpio;
+              escHardware[selected_esc].pin = pwmIOConfiguration->ioConfigurations[i].timerHardware->pin;
+              escHardware[selected_esc].pinpos = GetPinPos(escHardware[selected_esc].pin);
+              escHardware[selected_esc].gpio_config_INPUT.pin = escHardware[selected_esc].pin;
+              escHardware[selected_esc].gpio_config_INPUT.speed = Speed_2MHz; // see pwmOutConfig()
+              escHardware[selected_esc].gpio_config_INPUT.mode = Mode_IPU;
+              escHardware[selected_esc].gpio_config_OUTPUT = escHardware[selected_esc].gpio_config_INPUT;
+              escHardware[selected_esc].gpio_config_OUTPUT.mode = Mode_Out_PP;
+              ESC_INPUT;
+              ESC_SET_HI;
+              selected_esc++;
+          }
+      }
+    }
+    escCount = selected_esc;
+    selected_esc = 0;
+}
+
+void usb1WireDeInitializeVcp(void){
+    for (selected_esc = 0; selected_esc < (escCount); selected_esc++) {
+        escHardware[selected_esc].gpio_config_OUTPUT.mode = Mode_AF_PP; // see pwmOutConfig()
+        ESC_OUTPUT;
+        ESC_SET_LO;
+    }
+    escCount = 0;
+    pwmEnableMotors();
+}
+
+#define START_BIT_TIMEOUT_MS 2
+
+#define BIT_TIME (52)  //52uS
+#define BIT_TIME_HALVE (BIT_TIME >> 1) //26uS
+#define START_BIT_TIME (BIT_TIME_HALVE + 1)
+#define STOP_BIT_TIME ((BIT_TIME * 9) + BIT_TIME_HALVE)
+
+static void suart_putc_(uint8_t tx_b)
+{
+    uint32_t btime;
+    ESC_SET_LO; // 0 = start bit
+    btime = BIT_TIME + micros();
+    while (micros() < btime);
+    for(uint8_t bit = 0; bit <8; bit++)
+    {
+        if(tx_b & 1)
+        {
+            ESC_SET_HI; // 1
+        }
+        else
+        {
+            ESC_SET_LO; // 0
+        }
+        btime = btime + BIT_TIME;
+        tx_b = (tx_b >> 1);
+        while (micros() < btime);
+    }
+    ESC_SET_HI; // 1 = stop bit
+    btime = btime + BIT_TIME;
+    while (micros() < btime);
+}
+
+
+static uint8_t suart_getc_(uint8_t *bt)
+{
+    uint32_t btime;
+    uint32_t start_time;
+    uint32_t stop_time;
+    uint32_t wait_start;
+
+    *bt = 0;
+
+    wait_start = millis() + START_BIT_TIMEOUT_MS;
+    while (ESC_IS_HI) {
+        // check for start bit begin
+        if (millis() > wait_start) {
+            return 0;
+        }
+    }
+    // start bit
```

Comment:
Main advantage is that it is much easier to read ...

BTW: not important on 19200 (rounding error is ~0.1%) but it is possible to use fixed point BIT_TIME (for example 24.8 - multiply BIT_TIME by 256) and accumulate it into btime. Then use btime/256 (btime >> 8) when waiting.  This way rounding error will not accumulate, it will work fine even with few timer ticks (>5 is ok) per bit.